### PR TITLE
[Analytics Hub] Handle range selection granularity and interval

### LINKED ISSUE #8189: [Analytics Hub] Integrate UI and Yosemite layers to fetch real order stats data
We should make two stats requests for Analytics data: one for the current selected period and the previous selected period.

We have Yosemite support with `StatsActionV4.retrieveStats` to sync the `OrderStatsV4` data used for both the Revenue and Orders cards. This needs to be integrated with the Analytics Hub in the UI layer.

### PR DESCRIPTION
**Pull request by ThomazFB**
Closes #8189

Why
==========
The stats request are currently using a fixed granularity of days with an interval size of `31`. For better request performance and a correct chart render, we need to use the correct granularity and interval based on the selected time range.

How
==========
Adds to the `SelectedType` two computed properties: `granularity` and `intervalSize`, grouping the response based on the time range nature like `days`, `weeks`, `months`, `quarters` and `years`.

How to Test
==========
1. Open the Analytics Hub
2. Do an overall test verifying if the stats reflect what is displayed in the Woo Core Analytics section and if the charts are correctly rendered based on the stats response numbers.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift [hunk 1]
```diff
@@ -104,6 +104,41 @@ extension AnalyticsHubTimeRangeSelection {
...
+        var intervalSize: Int {
+            switch self {
+            case .today, .yesterday:
+                return 24
+            case .weekToDate, .lastWeek:
+                return 7
+            case .monthToDate, .lastMonth:
+                return 31
+            case .quarterToDate, .lastQuarter:
+                return 12
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift]: For amount of weeks in a quarter should we pick 13 to be more precise?
Year is 52 weeks (+1-2 days), so 52+/4 = 13+

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift [hunk 2]
```diff
@@ -104,6 +104,41 @@ extension AnalyticsHubTimeRangeSelection {
...
+            case .monthToDate, .lastMonth:
+                return .daily
+            case .quarterToDate, .lastQuarter:
+                return .weekly
+            case .yearToDate, .lastYear:
+                return .monthly
+            }
+        }
+
+        /// The resopnse interval size that should be used to request stats from the given SelectedType
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift]: nit:
```suggestion
        /// The response interval size that should be used to request stats from the given SelectedType
```

**Review by ealeksandrov — APPROVED** [review]
LGTM with one small improvement suggestion! :shipit:

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift (hunk 1 above)
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift]: Makes perfect sense! Thanks! Fixed here: https://github.com/woocommerce/woocommerce-ios/pull/8341/commits/1ca89c255dabe141e614ca54ae36026ba8436049

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift (hunk 2 above)
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift]: Thanks! Fixed here: https://github.com/woocommerce/woocommerce-ios/pull/8341/commits/1ca89c255dabe141e614ca54ae36026ba8436049

**Pull request merged**